UserDocs.md

```diff
@@ -14,245 +14,233 @@ You may wish create a Quilt-specific [profile](https://docs.aws.amazon.com/cli/l
 
-#### `helium.config(URL)`
-Set the local configuration from the URL of a T4 instance.  Overwrites all local config 
-values if successful, but keeps a backup in the config dir.
 
-#### `helium.config(KEY=VALUE [, KEY2=VALUE2, ...])`
-Manually sets specific configuration options.
+<!-- Note that in order for `get` to work, the file must have been created by t4.put and it must be in one of the formats supported by the default T4 deserializers. -->
 
-### Navigation
 
-#### `helium.search(key)`
-Searches a T4 bucket for `key`. Returns a list of search results.
+### Moving data not on T4
+We encourage a push/pull architecture based on having data packages ultimately live on T4. But you may also use the t4 CLI to handle moving data anywhere on your local file system.
 
-In order for `search` to work, you must use `he.configure()` to
-point T4 to an ElasticSearch endpoint that indexes your
-T4 bucket.
+For example, suppose that you have NAS accessible on your machine from `"nas://"`. All of constructing, pushing, and pulling still work.
 
-See [Metadata and search](#metadata-and-search) for additional syntax.
+To create a package referencing that data and save it locally:
 
-#### `helium.ls(path)`
+    p = t4.Package()
+    p = p.set("foo", "nas://foo")
+    p.build("example-package")
 
-Lists the contents of a path in a T4 bucket.
-<!-- TODO: for all return values, show output instead of describing it -->
-Returns a tuple whose first item is a list of sub-paths,
-and whose second item is metadata about the first item.
-Each version of an object in S3 gets its own entry in the list.
+To load the contents of that package locally:
 
-## User guide
+	p.copy("/", "target/directory")
 
-### Working with memory
 
-You can commit a Python object to T4 using the `put` command:
+Packages pushed to T4 guarantee the immutability of constituent nodes, so long as the underlying S3 bucket has versioning enabled. Packages materialized elsewhere do not. The onus of keeping the package manifest consistent with the actual contents of the file (e.g. updating the package manifest every time the file changes) is on the package author. Additionally, even if package files get overwritten, there is no way of accessing the old files.
 
-```python
-# Generate example data
-import pandas as pd
-import numpy as np
-df = pd.DataFrame(np.random.random((1000, 10)))
-
-# Put it
-import helium as he
-he.put(df, "bucket-name/my-frame.parquet")
-```
 
-The above code writes `df` at the top level of `bucket-name`.
+### Updating a package
+Updating a package in a registry is a matter of pushing to it again with modified contents. For example:
 
-> The `.parquet` file extension is for reference. T4 uses object metadata to store the object's format, and can deserialize the dataframe with or without the file extension 
+	# push an initial empty package to a T4 bucket
+	p = t4.Package()
+	p.build("example-package")
+	p.push("example-package", "s3://my-t4-bucket")
```

Comment:
use subdirectories in the examples so users aren't pushing to the top of their buckets -- pushes are hard to undo